src/cpp/core/r_util/RSessionContext.cpp

```diff
@@ -207,12 +211,23 @@ bool isSharedPath(const std::string& projectPath,
       if (st.st_gid == user.getGroupId())
          return false;
 
-#ifndef __APPLE__
       // not shared if we're in any of the groups that own the directory
       // (note that this checks supplementary group IDs only, so the check
       // against the primary group ID above is still required)
+#ifdef HAVE_GROUP_MEMBER
       if (::group_member(st.st_gid))
          return false;
+#else
+      // this is basically what glibc, gnulib, and glibcompat do to implement
+      // group_member()
+      gid_t groups[NGROUPS_MAX];
+      int ngroups = ::getgroups(NGROUPS_MAX, groups);
```

Comment:
Should we check for and log errors here (if `::getgroups()` returns `-1`)?

Reply:
(it seems like that should never happen but better safe than sorry IMHO)

Reply:
Yep, makes sense.